requirements.txt

```diff
@@ -1 +1,2 @@
 six>=1.7.0,<=1.15.0
+contextvars~=2.4; python_version >= "3.0" and python_version < "3.7"
```

Comment:
As a note, this library is stable, and has been on version 2.4 for the last 4 years.

Reply:
Got it, that's good. I worry about adding the dependency cause it could cause a dependencies resolving problem with other libraries. What if we remove it from the main dependencies and modify the guard clause in the provider to something like this:

```
        if not contextvars:
            raise RuntimeError(
                'Contextvars library not found. This provider '
                'requires Python 3.7 or a backport of contextvars. '
                'To install a backport run "pip install contextvars".'
            )
```

How does it look to you?

Reply:
Sure thing, no objections.  I'm not really concerned about people on older Python versions.

However we could probably get rid of the `~=2.4` version constraint and still be fine.  Considering just how old that library is.